### ISSUE TITLE
Simplify access to content's HTML for plugins

### ISSUE THREAD

**Issue by honzajavorek** [getpelican/pelican#1921]
From what I've seen in the wild, what most of the plugins do is they modify the output HTML before it gets written. They listen on

```python
def register():
    signals.content_object_init.connect(...)
```

...and they use `content._content`, which seems like hack - using private property directly like this. Chaining of plugins is also AFAIK officially undefined, which isn't helpful for a plugin creator and/or user. According to my error-and-trials, it seems that plugins defined earlier in the `PLUGINS` list get executed later:

```python
# The order is significant! [featured_image, media] makes sure that media
# gets executed before featured_image.
PLUGINS = [
    code_blocks, heading_permalinks, plain_summary, featured_image, media
]
```

But it can be also just me or some coincidence. I didn't look into Pelican's code. At least we could put it in the docs if my observations are right.

Also, I'm not sure whether it's documented at which moment the `article.summary` gets filled. Some plugins change the content, but the changes won't get reflected in the summary. Should they alter summary too or should they just listen on a different signal?

---------

So to sum up,  seems like it's not very pleasant experience and it feels like hack of the plugin system. But since what most of the plugins actually do is that they touch the HTML, I think it would be better to improve the experience, either by making content modifications more accessible or by improving docs.

**Comment by ingwinlu** [comment]
your observations are wrong. the order is unspecified (https://pythonhosted.org/blinker/ send/2).

accessing _content is not a hack. accessing .content invokes a lot of things that you might not want. having both is allowing plugin designers to do choose. (and presents a nice interface to people who only want to do themes).

We kind of expect plugin developers to look into the pelican source code. It documents what is happening better then we could ever describe in the docs.

**Comment by honzajavorek** [comment]
Thanks for clarifications! I always felt about `content._content` as "this property can disappear tomorrow and if you are using it, you know what you are doing", not as "this is secret API for plugin developers and if you happen to be one, feel free to use it". I think that's the main reason I filed this issue.
